external/gtest/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# We need thread support
+find_package(Threads REQUIRED)
+
+# Enable ExternalProject CMake module
+include(ExternalProject)
+
+# Download and install GoogleTest
+ExternalProject_Add(
```

Comment:
Think about [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) - a "git way" to add external dependencies.